docs/docs/canonicai/chapter3.md

```diff
@@ -0,0 +1,339 @@
+# Build a conversational AI system in Jaseci
+
+# Chapter 3
+
+### **cai_state node**
+### Two main stages in handling a conversational AI request
+![Two Stages](./images/nlu_nlg.png?raw=true "Title")
```

Comment:
For the diagram, can we annotate the arrows with what is the input and output of each module? For example, input to intent classificaiton would be the input text and list of candidate intent labels and output would be the predicted intent.

Reply:
Like an example? I'm kinda confused.

Reply:
No. Just annotate the arrows with what the input and output is. Like, literally "input text" and "candidate intent labels"

Reply:
Okay this is moving in the right direction. Although, technically what's currently on the diagram is not correct...."candidate labels" are input to intent classification instead of the output.

* intent classification: input is "input text" and "candidate labels", output is "predicted intent".
* entity extraction: input is "input text", output is "extracted entities."

Can you update?